cli/sdo/voucher.go

```diff
@@ -151,6 +151,126 @@ type ImportResponse struct {
 	NodeToken string `json:"nodeToken"`
 }
 
+// call GET sdoURL/vouchers/[<device-uuid>] to get the uploaded vouchers 
+func getVouchers (org, userCreds, apiMsg string, voucher string) ([]byte, string) {
+	msgPrinter := i18n.GetMessagePrinter()
+	cliutils.Verbose(msgPrinter.Sprintf("Listing imported SDO vouchers."))
+
+	// setup HTTP parameters and URL
+	var respBodyBytes []byte
+	var requestBodyBytes []byte
+	var sdoURL string
+	url := cliutils.GetSdoSvcUrl()
+	
+	if voucher == "" {
+		sdoURL = url+"/vouchers"
+	} else {
+		sdoURL = url+"/vouchers"+"/"+voucher 
+	}
+
+	creds := cliutils.OrgAndCreds(org, userCreds)
+	method := http.MethodGet
+	apiMsg = method + " " + sdoURL
+	httpClient := cliutils.GetHTTPClient(config.HTTPRequestTimeoutS)
+
+	resp := cliutils.InvokeRestApi(httpClient, method, sdoURL, creds, requestBodyBytes, "SDO Owner Service", apiMsg)
+	defer resp.Body.Close()
+	httpCode := resp.StatusCode
+	cliutils.Verbose(msgPrinter.Sprintf("HTTP code: %d", httpCode))
+
+	respBodyBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		cliutils.Fatal(cliutils.HTTP_ERROR, msgPrinter.Sprintf("failed to read exchange body response from %s: %v", apiMsg, err))
+	}
+	if httpCode != 200 {
+		cliutils.Fatal(cliutils.HTTP_ERROR, msgPrinter.Sprintf("bad HTTP code %d from %s: %s", httpCode, apiMsg, string(respBodyBytes)))
+	}
+	if err != nil {
+		cliutils.Fatal(cliutils.HTTP_ERROR, msgPrinter.Sprintf("json unmarshalling HTTP response '%s' from %s: %v", string(respBodyBytes), apiMsg, err))
+	}
+
+	return respBodyBytes, apiMsg
+}
+
+// called when the -l flag is used to list the full voucher
+func listFullVoucher (respBodyBytes []byte, apiMsg string) []byte {
+	msgPrinter := i18n.GetMessagePrinter()
+	cliutils.Verbose(msgPrinter.Sprintf("Listing imported SDO vouchers."))
+
+	var output interface{}
```

Comment:
Would be nice to have a comment explaining why you are unmarshalling into `interface{}` and not a struct like usual.

Reply:
Ah yes, good idea.